src/client.rs

```diff
@@ -16,26 +16,13 @@ pub struct CadreClient {
 impl CadreClient {
     /// Create a new client object pointing at a given HTTP origin.
     #[allow(clippy::unnecessary_unwrap)]
-    pub fn new(origin: Option<&str>) -> Self {
+    pub fn new(origin: &str) -> Self {
         let mut connector = HttpConnector::new();
         connector.set_nodelay(true);
 
-        // use env var for overriding default origin
-        let origin_value: String = if origin.is_none() {
-            let cadre_url = match option_env!("CADRE_URL") {
```

Comment:
Can we use `std::env::var()` to inject this environment variable? It should be the cadre client the one sensitive to it, no?

Reply:
Yes, the correct way to do this would be to use `std::env::var()`. No, it should not be set in library code. This is a side-effect. Environment variables can be passed as configuration, read in `main` or using `clap`.